src/store/index.tsx

```diff
@@ -36,8 +37,14 @@ export const StoreContext = createContext<StoreContextType>(
   {} as StoreContextType
 );
 
-export const StoreProvider = ({ children }: StoreProviderProps) => {
-  const [store, dispatchAction] = useReducer(reducer, initialStore);
+export const StoreProvider = ({
+  children,
+  initialStore,
+}: StoreProviderProps) => {
+  const [store, dispatchAction] = useReducer(
+    reducer,
+    initialStore || defaultInitialStore
+  );
```

Comment:
I'm not sure to understand these changes 🤔 
Could you provide more insights?

Thanks! :)

Reply:
This one is for the testing purpose of setting the default initial store. I tried stub the `const initialStore`, but it is pretty tricky to do because it has to change the way to export the const to make it stub-able and needs to reset the stub for every test that mocks the `initialStore`.

So I prefer to inject the default store by prop is more straightforward to do, and this component is not widely used. 💭

Reply:
ok, perfect! Thanks for the explanation :) All good if it is for testability (stub-ability) :)